engines/director/debugger.cpp

```diff
@@ -34,22 +36,151 @@ Debugger *g_debugger;
 
 Debugger::Debugger(): GUI::Debugger() {
 	g_debugger = this;
-	registerCmd("lingo", WRAP_METHOD(Debugger, cmd_lingo));
+	registerCmd("help", WRAP_METHOD(Debugger, cmdHelp));
+
+	registerCmd("version", WRAP_METHOD(Debugger, cmdVersion));
+
+	registerCmd("repl", WRAP_METHOD(Debugger, cmdRepl));
+	registerCmd("stack", WRAP_METHOD(Debugger, cmdStack));
+	registerCmd("st", WRAP_METHOD(Debugger, cmdStack));
+	registerCmd("frame", WRAP_METHOD(Debugger, cmdFrame));
+	registerCmd("f", WRAP_METHOD(Debugger, cmdFrame));
+	registerCmd("backtrace", WRAP_METHOD(Debugger, cmdBacktrace));
+	registerCmd("bt", WRAP_METHOD(Debugger, cmdBacktrace));
+	registerCmd("vars", WRAP_METHOD(Debugger, cmdVars));
+	registerCmd("step", WRAP_METHOD(Debugger, cmdStep));
+	registerCmd("s", WRAP_METHOD(Debugger, cmdStep));
+	registerCmd("next", WRAP_METHOD(Debugger, cmdNext));
+	registerCmd("n", WRAP_METHOD(Debugger, cmdNext));
+	registerCmd("finish", WRAP_METHOD(Debugger, cmdFinish));
+	registerCmd("fin", WRAP_METHOD(Debugger, cmdFinish));
+
+	_step = false;
+	_stepCounter = 0;
+	_finish = false;
+	_finishCounter = 0;
+	_next = false;
+	_nextCounter = 0;
 }
 
 Debugger::~Debugger() {
 	if (_out.isOpen())
 		_out.close();
 }
 
-bool Debugger::cmd_lingo(int argc, const char **argv) {
-	if (argc == 2 && !strcmp(argv[1], "on")) {
-		registerDefaultCmd(WRAP_DEFAULTCOMMAND(Debugger, lingoCommandProcessor));
-		debugPrintf(PROMPT);
-	}
+bool Debugger::cmdHelp(int argc, const char **argv) {
+	debugPrintf("\n");
+	debugPrintf("Debug flags\n");
+	debugPrintf("-----------\n");
+	debugPrintf("debugflag_list - Lists the available debug flags and their status\n");
+	debugPrintf("debugflag_enable - Enables a debug flag\n");
+	debugPrintf("debugflag_disable - Disables a debug flag\n");
+	debugPrintf("debuglevel - Shows or sets debug level\n");
+	debugPrintf("\n");
+	debugPrintf("Commands\n");
+	debugPrintf("--------\n");
+	debugPrintf("Player:\n");
+	debugPrintf(" version - Shows the Director version\n");
+	//debugPrintf(" movie [moviePath] - Get or sets the current movie\n");
+	//debugPrintf(" movieinfo - Show information for the current movie\n");
+	//debugPrintf(" scoreframe [frameNum] - Gets or sets the current score frame\n");
+	//debugPrintf(" channels [frameNum] - Shows channel information for a score frame\n");
+	//debugPrintf(" cast - Shows the cast list for the current movie\n");
+	debugPrintf("\n");
+	debugPrintf("Lingo execution:\n");
+	//debugPrintf(" eval [statement] - Evaluates a single Lingo statement\n");
+	debugPrintf(" repl - Switches to a REPL interface for evaluating Lingo code\n");
+	debugPrintf(" backtrace / bt - Prints a backtrace of all stack frames\n");
+	//debugPrintf(" disasm [function] - Lists the bytecode disassembly for a script function\n");
+	debugPrintf(" stack / st - Lists the elements on the stack\n");
+	debugPrintf(" frame / f - Prints the current script frame\n");
```

Comment:
Please rename `frame`. It conflicts with the common use of frame for a frame in a scene.